plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiMetadata.java

```diff
@@ -291,7 +291,7 @@ public void validateScan(ConnectorSession session, ConnectorTableHandle handle)
     public boolean allowSplittingReadIntoMultipleSubQueries(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         // dont split to subqueries if tableHandle is systemTableHandle, hudi supports only a columnar (parquet) storage format
```

Comment:
If this PR is mostly reverting 2be471bb25ca4448b2fd280ff9ce104220c3167f then this comment needs to be updated as well :)

Reply:
It moves check from Metadata to MetadataManager as it could potentially cause problem for other connectors.

Reply:
Sure, but the comment is outdated now.

```suggestion
        // hudi supports only a columnar (parquet) storage format
```